Objects/sliceobject.c

```diff
@@ -148,6 +148,37 @@ PySlice_New(PyObject *start, PyObject *stop, PyObject *step)
     return (PyObject *) obj;
 }
 
+PyObject *
+_PyBuildSlice_Consume2(PyObject *start, PyObject *stop)
+{
+    assert(start != NULL && stop != NULL);
+
+    PyInterpreterState *interp = _PyInterpreterState_GET();
+    PySliceObject *obj;
+    if (interp->slice_cache != NULL) {
+        obj = interp->slice_cache;
+        interp->slice_cache = NULL;
+        _Py_NewReference((PyObject *)obj);
+    }
+    else {
+        obj = PyObject_GC_New(PySliceObject, &PySlice_Type);
+        if (obj == NULL) {
+            goto error;
+        }
+    }
```

Comment:
This looks like a copy of lines 127-138. Should it be extracted to a helped function?